src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -61,7 +61,7 @@ private static EqualityComparer<T> CreateComparer()
             }
             // See the METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST and METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST_LONG cases in getILIntrinsicImplementation
             else if (t.IsEnum) {
-                TypeCode underlyingTypeCode = Type.GetTypeCode(Enum.GetUnderlyingType(t));
+                TypeCode underlyingTypeCode = Type.GetTypeCode(t);
```

Comment:
@AlexGhiondea As far as I can see this will always have the same effects as if `Enum.GetUnderlyingType` was excluded: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/RtType.cs#L3400

However, it looks like explicitly calling `GetUnderlyingType` can avoid another P/Invoke + virtual method call (which would degrade performance), so I'll change this back for now.